### ISSUE TITLE
Unisoc Devices headphones error 

### ISSUE THREAD

**Issue by Claravallac** [phhusson/treble_experimentations#2526]
Headphones are not recognized on Unisoc Chips in AOSP 12+ 

(Bluetooth headphones work normally)

**Comment by Reyproject** [comment]
Can confirm this as I also have the same chip (Unisoc T610). Neither the Use Alternate nor the command doesn't fix it.

**Comment by Claravallac** [comment]
@Reyproject It looks like this is a general problem with unisoc ones, mine is a T700

**Comment by Reyproject** [comment]
@Claravallac Yeah, I've only tried LineageOS though. Will still try to install other GSIs since SafetyNet doesn't pass for me. If the problem persists on different GSIs then I'll try Android 11. If it's still there I'm going back to stock.

**Comment by Reyproject** [comment]
Still on Lineage 19.1. I noticed that even though the headset is plugged in, sound does not go there and remains on phone speakers. But using the audio controls (Play/Pause, Next, Back) works. Only needs to pass the audio it which I don't know how.

**Comment by Reyproject** [comment]
@Claravallac have you tried other GSIs as well?

**Comment by Claravallac** [comment]
@Reyproject I've used several different GSIs but never had any major problems with "safetynet"

**Comment by Claravallac** [comment]
call me on telegram "@ClaraVallac"

**Comment by Reyproject** [comment]
Sent you a message just now.

**Comment by Reyproject** [comment]
I tried Android 13 GSIs. They all make the camera and video playback green. Although when taking a photo everything is normal. Meanwhile video is the same anywhere whether it's on YouTube, MX Player or browsers

**Comment by Reyproject** [comment]
Also, on Android 12+, microphone isn't functioning properly. Either it's very low or no sound at all

**Comment by CarlosCMD** [comment]
Am also on lineage OS 20. The wired headphones works but still the speaker work. Am on sc9832e.
About the videos, mine shows up broken thumbnails.

**Comment by phhusson** [comment]
Hello

Please dump the result of `adb exec-out dumpsys media.audio_policy` and `tinymix` (to be run as root. Either over adb or termux or whatever app), in those 4 cases:

- A11, no headphone connected
- A11 headphones connected
- A12/A13/A14 (whichever you want), no headphone connected
- A12/A13/A14 headphone connected

**Comment by fanuelberhanukebe** [comment]
[android 11 no headphone audio policy.txt](https://github.com/phhusson/treble_experimentations/files/13268577/android.11.no.headphone.audio.policy.txt)
[android 11 with headphone audio policy.txt](https://github.com/phhusson/treble_experimentations/files/13268581/android.11.with.headphone.audio.policy.txt)
[android 13 no headphone audio policy.txt](https://github.com/phhusson/treble_experimentations/files/13268582/android.13.no.headphone.audio.policy.txt)
[android 13 with headphone audio policy.txt](https://github.com/phhusson/treble_experimentations/files/13268583/android.13.with.headphone.audio.policy.txt)

**Comment by fanuelberhanukebe** [comment]
> Hello
> 
> Please dump the result of `adb exec-out dumpsys media.audio_policy` and `tinymix` (to be run as root. Either over adb or termux or whatever app), in those 4 cases:
> 
> * A11, no headphone connected
> * A11 headphones connected
> * A12/A13/A14 (whichever you want), no headphone connected
> * A12/A13/A14 headphone connected

[android 11 no headphone tinymix.txt](https://github.com/phhusson/treble_experimentations/files/13269810/android.11.no.headphone.tinymix.txt)
[android 11 with headphone tinymix.txt](https://github.com/phhusson/treble_experimentations/files/13269812/android.11.with.headphone.tinymix.txt)
[android 13 no headphone tinymix.txt](https://github.com/phhusson/treble_experimentations/files/13269814/android.13.no.headphone.tinymix.txt)
[android 13 with headphone tinymix.txt](https://github.com/phhusson/treble_experimentations/files/13269815/android.13.with.headphone.tinymix.txt)

**Comment by PMorty282** [comment]
> Can confirm this as I also have the same chip (Unisoc T610). The Use Alternate nor the command doesn't fix it.

Same problem here, with the same chipset. Is there no fix to this?

**Comment by sssunn** [comment]
Using Corvus os 4.1 A12L
Unisoc T610
Realme c25y A11
Wired Headphones are not being detected. 
All A13 and plus gsi is having video playback issues.

**Comment by PMorty282** [comment]
> I tried Android 13 GSIs. They all make the camera and video playback green. Although when taking a photo everything is normal. Meanwhile video is the same anywhere whether it's on YouTube, MX Player or browsers

Same here as well. Issue persists in Android 14 GSIs as well. Any camera, video playback, and if I can remember, GIF and image playback, has this issue. Also still without no audio. I really hope someone could fix this.
